probe-rs/src/probe.rs

```diff
@@ -674,8 +673,8 @@ pub struct DebugProbeInfo {
     pub product_id: u16,
     /// The serial number of the debug probe.
     pub serial_number: Option<String>,
-    /// The probe type of the debug probe.
-    pub probe_type: &'static dyn ProbeFactory,
+    /// A reference to the [`ProbeFactory`] that created this info object.
+    pub probe_factory: &'static dyn ProbeFactory,
```

Comment:
I think this should be private, otherwise people will depend on it.

Reply:
What about `hid_interface` which looks to be unused?